lib/http1.c

```diff
@@ -451,8 +453,8 @@ static const char *fixup_request(struct st_h2o_http1_conn_t *conn, struct phr_he
     }
 
     /* init headers */
-    if ((ret = init_headers(&conn->req.pool, &conn->req.headers, headers, num_headers, &connection, &host, &upgrade, expect,
-                            entity_header_index)) != NULL)
+    if ((ret = init_headers(&conn->req.pool, &conn->req.headers, headers, num_headers, &connection, &host, &upgrade,
+                            expect, expect_orig_case, entity_header_index)) != NULL)
```

Comment:
Rather than passing the name and value separately, how about passing the index of the expect header within `headers`?

Then, we can get the name by using `headers[expect_index].name(_len)` and the value by `headers[expect_index].value(_len)`.

Reply:
Addressed in https://github.com/h2o/h2o/pull/3419/commits/01fc92a93cf4d8b6f6e34aa65d6a0c10e929a27a